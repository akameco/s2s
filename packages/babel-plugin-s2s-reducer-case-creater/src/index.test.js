// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const getFixturesPath = x => path.resolve(__dirname, '__fixtures__', x)
const fromPath = getFixturesPath('actionTypes.js')
const noopPath = getFixturesPath('noop.js')
const withParamsPath = getFixturesPath('sameParam.js')

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  pluginOptions: { from: fromPath },
  tests: [
    {
      title: 'basic',
      code: `
export default function(state: State = initialState, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return { ...state, count: state + 1 }

    default:
      return state
  }
}
      `,
    },
    {
      title: 'not Actions',
      code: `
export default function(state: State = initialState, action) {
  switch (action.type) {
    case Other.INCREMENT:
      return { ...state, count: state + 1 }
    case Actions.INCREMENT:
      return { ...state, count: state + 1 }
    default:
      return state
  }
}
      `,
    },
    {
      title: 'basic',
      code: `
function update(state: State = initialState, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return { ...state, count: state + 1 }
    default:
      return state
  }
}

export default function(state: State = initialState, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return update(state, action)
    default:
      return state
  }
}
      `,
    },
  ],
})

pluginTester({
  title: 'when actionTypes.js is empty',
  plugin,
  snapshot: true,
  pluginOptions: { from: noopPath },
  tests: [
    `
export default function(state: State = initialState, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return { ...state, count: state + 1 }

    default:
      return state
  }
}
      `,
  ],
})

pluginTester({
  plugin,
  tests: [
    {
      title: 'throw error',
      code: `// throw error`,
      error: /required from option/,
    },
  ],
})

pluginTester({
  title: 'when has same property',
  plugin,
  snapshot: true,
  pluginOptions: { from: withParamsPath },
  tests: [
    `
export default function(state: State = initialState, action) {
  switch (action.type) {
    default:
      return state
  }
}
      `,
  ],
})
