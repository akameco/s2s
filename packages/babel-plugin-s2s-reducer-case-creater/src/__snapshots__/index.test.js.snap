// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default basic: basic 1`] = `
"
export default function(state: State = initialState, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return { ...state, count: state + 1 }

    default:
      return state
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

export default function (state: State = initialState, action) {
  switch (action.type) {
    case Actions.DECREMENT:
      return {
        ...state
      };

    case Actions.INCREMENT:
      return { ...state, count: state + 1 };

    default:
      return state;
  }
}
"
`;

exports[`default basic: basic 2`] = `
"
function update(state: State = initialState, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return { ...state, count: state + 1 }
    default:
      return state
  }
}

export default function(state: State = initialState, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return update(state, action)
    default:
      return state
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

function update(state: State = initialState, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return { ...state, count: state + 1 };
    default:
      return state;
  }
}

export default function (state: State = initialState, action) {
  switch (action.type) {
    case Actions.DECREMENT:
      return {
        ...state
      };

    case Actions.INCREMENT:
      return update(state, action);
    default:
      return state;
  }
}
"
`;

exports[`default not Actions: not Actions 1`] = `
"
export default function(state: State = initialState, action) {
  switch (action.type) {
    case Other.INCREMENT:
      return { ...state, count: state + 1 }
    case Actions.INCREMENT:
      return { ...state, count: state + 1 }
    default:
      return state
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

export default function (state: State = initialState, action) {
  switch (action.type) {
    case Actions.DECREMENT:
      return {
        ...state
      };

    case Other.INCREMENT:
      return { ...state, count: state + 1 };
    case Actions.INCREMENT:
      return { ...state, count: state + 1 };
    default:
      return state;
  }
}
"
`;

exports[`when actionTypes.js is empty 1. s2s-reducer-case-creater: 1. s2s-reducer-case-creater 1`] = `
"
export default function(state: State = initialState, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return { ...state, count: state + 1 }

    default:
      return state
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

export default function (state: State = initialState, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return { ...state, count: state + 1 };

    default:
      return state;
  }
}
"
`;

exports[`when has same property 1. s2s-reducer-case-creater: 1. s2s-reducer-case-creater 1`] = `
"
export default function(state: State = initialState, action) {
  switch (action.type) {
    default:
      return state
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

export default function (state: State = initialState, action) {
  switch (action.type) {
    case Actions.SET:
      return {
        ...state,
        count: action.count
      };

    default:
      return state;
  }
}
"
`;
